src/services/claim.ts

```diff
@@ -0,0 +1,182 @@
+  }
+};
+
+export async function getSnapshot(network: NetworkId) {
+  if (constants[network]?.snapshot) {
+    return (await ipfsGet(gateway, constants[network].snapshot, 'ipns')) || {};
+  }
+  return {};
+}
+
+type ClaimStatus = boolean;
+
+export async function getClaimStatus(
+  network: NetworkId,
+  provider: Web3Provider,
+  ids: number,
+  account: string
+): Promise<ClaimStatus[]> {
+  return await call(provider, merkleRedeemAbi, [
+    constants[network].merkleRedeem,
+    'claimStatus',
+    [account, 1, ids]
+  ]);
+}
+
+export type Report = Record<string, any>;
+
+export async function getReports(snapshot: Snapshot, weeks: number[]) {
+  const reports = await Promise.all<Report>(
+    weeks.map(week => ipfsGet(gateway, snapshot[week]))
+  );
+  return Object.fromEntries(reports.map((report, i) => [weeks[i], report]));
+}
+
+export async function getPendingClaims(
+  network: NetworkId,
+  provider: Web3Provider,
+  account: string
+): Promise<{ claims: Claim[]; reports: Report }> {
+  const snapshot = await getSnapshot(network);
+
+  const claimStatus = await getClaimStatus(
+    network,
+    provider,
+    Object.keys(snapshot).length,
+    account
+  );
+
+  const pendingWeeks = claimStatus
+    .map((status, i) => [i + 1, status])
+    .filter(([, status]) => !status)
+    .map(([i]) => i) as number[];
+
+  const pendingWeeksReports = await getReports(snapshot, pendingWeeks);
+
+  return {
+    claims: Object.entries(pendingWeeksReports)
+      .filter((report: Report) => report[1][account])
+      .map((report: Report) => {
+        return {
+          id: report[0],
+          amount: report[1][account],
+          amountDenorm: parseUnits(report[1][account], 18)
+        };
+      }),
+    reports: pendingWeeksReports
+  };
+}
+
+type CurrentRewardsEstimateResponse = {
+  success: boolean;
+  result: {
+    current_timestamp: string;
+    'liquidity-providers': Array<{
+      snapshot_timestamp: string;
+      address: string;
+      current_estimate: string;
+      velocity: string;
+      week: number;
+    }>;
+  };
+};
+
+export type CurrentRewardsEstimate = {
+  rewards: string;
+  velocity: string;
+  timestamp: string;
+} | null;
+
+export async function getCurrentRewardsEstimate(
+  account: string
+): Promise<CurrentRewardsEstimate> {
+  try {
+    const response = await axios.get<CurrentRewardsEstimateResponse>(
+      `https://api.balancer.finance/liquidity-mining/v1/liquidity-provider/${account}`
+    );
+    if (response.data.success) {
+      const liquidityProviders = response.data.result['liquidity-providers'];
+
+      if (Array.isArray(liquidityProviders)) {
+        const lastEntry = liquidityProviders[liquidityProviders.length - 1];
+
+        return {
+          rewards: liquidityProviders
+            .reduce(
+              (total, { current_estimate }) => total.plus(current_estimate),
+              bnum(0)
+            )
+            .toString(),
+          velocity: lastEntry.velocity,
```

Comment:
We can't count on the lastEntry being the current week, but we can count on one of the entries always having a `velocity` of zero. So IIUC we can sum the `velocity`s like you did with the `current_estimate`s